src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral2.java

```diff
@@ -54,13 +58,38 @@ private void p() {
         }
     }
 
+    public class MyResource implements AutoCloseable {
+        public void close() throws Exception {
+            System.out.println("Closing!");
+        }
+    }
+
     void method() throws Exception {
-        // Until https://github.com/checkstyle/checkstyle/issues/11968
-        String a = ""; // violation 'Distance between .* declaration and its first usage is 2.'
-        try (AutoCloseable i = new java.io.StringReader(a)) {
+        byte[] buf=new byte[10]; // violation 'Distance .* is 3.'
+
+        File b = new File("");
+        try(MyResource myResource= new MyResource();
+            InputStream finstr = Files.newInputStream(b.toPath());
+            BufferedInputStream in = new BufferedInputStream(finstr,buf.length)){
+            int length;
+            while((length=in.read(buf))!=-1){
+                System.out.println(buf.toString());
+            }
+        }
+    }
+
+    void methodTry() {
+        File a = new File(""); // violation 'Distance .* is 2.'
+        p();
+        p();
```

Comment:
For all examples, don't have same-name method calls in between. If you use the same name method calls, they are considered initialization. Considering it as initialization is broken and it is a known bug and it will lead to wrong test results. PR for this is already open https://github.com/checkstyle/checkstyle/pull/11672